src/main/java/ecdar/presentations/Grid.java

```diff
@@ -62,12 +62,12 @@ public void handleTranslateY(double oldValue, double newValue, double scale) {
 
     /**
      * Redraw the grid in center of the screen
-     * @param scale the scale in which to draw the grid
+     * @param zoomLevel the scale in which to draw the grid
      */
-    public void updateGrid(double scale) {
-        // The given size of the canvas divided by the given scale
-        double screenWidth = (int) Grid.snap(((CanvasShellPresentation) getParent().getParent()).getWidth() / scale * 4);
-        double screenHeight = (int) Grid.snap(((CanvasShellPresentation) getParent().getParent()).getHeight() / scale * 4);
+    public void updateGrid(double zoomLevel) {
+        // The given size of the canvas divided by the given zoomLevel (multiplied by 6 to ensure that the grid covers the full canvas)
+        double screenWidth = (int) Grid.snap(((CanvasShellPresentation) getParent().getParent()).getWidth() / zoomLevel * 6);
+        double screenHeight = (int) Grid.snap(((CanvasShellPresentation) getParent().getParent()).getHeight() / zoomLevel * 6);
```

Comment:
The multiplier will be replaced with a more sophisticated solution in a later PR